File: kirun-js/__tests__/engine/repository/RepositoryFilterTest.ts
import { KIRunSchemaRepository, KIRunFunctionRepository } from '../../../src';

const funcRepo = new KIRunFunctionRepository();
const schemaRepo = new KIRunSchemaRepository();

test('Repository Filter Test', async () => {
    expect(await funcRepo.filter('Rep')).toStrictEqual([
        'System.String.Repeat',
        'System.String.Replace',
        'System.String.ReplaceFirst',
        'System.String.PrePad',
        'System.String.ReplaceAtGivenPosition',
    ]);

    expect(await funcRepo.filter('root')).toStrictEqual([
        'System.Math.CubeRoot',
        'System.Math.SquareRoot',
    ]);

    expect(await schemaRepo.filter('root')).toStrictEqual([]);

    expect(await schemaRepo.filter('rin')).toStrictEqual(['System.string']);

    expect(await schemaRepo.filter('ny')).toStrictEqual(['System.any']);

    expect(await schemaRepo.filter('')).toStrictEqual([
        'System.any',
        'System.boolean',
        'System.double',
        'System.float',
        'System.integer',
        'System.long',
        'System.number',
        'System.string',
        'System.ParameterExpression',
        'System.Null',
        'System.Schema',
    ]);
});

File: kirun-java/src/test/java/com/fincity/nocode/kirun/engine/repository/RepositoryFilterTest.java
package com.fincity.nocode.kirun.engine.repository;

import java.util.Set;
import java.util.stream.Collectors;

import org.junit.jupiter.api.Test;

import com.fincity.nocode.kirun.engine.repository.reactive.KIRunReactiveFunctionRepository;
import com.fincity.nocode.kirun.engine.repository.reactive.KIRunReactiveSchemaRepository;

import reactor.test.StepVerifier;

class RepositoryFilterTest {

	@Test
	void test() {

		var funcRepo = new KIRunReactiveFunctionRepository();
		var schemaRepo = new KIRunReactiveSchemaRepository();

		StepVerifier.create(funcRepo.filter("Rep").collect(Collectors.toSet()))
				.expectNext(Set.of("System.String.PrePad", "System.String.ReplaceAtGivenPosition",
						"System.String.ReplaceFirst",
						"System.String.Repeat", "System.String.Replace"))
				.verifyComplete();

		StepVerifier.create(funcRepo.filter("root").collect(Collectors.toSet()))
				.expectNext(Set.of("System.Math.CubeRoot", "System.Math.SquareRoot"))
				.verifyComplete();

		StepVerifier.create(schemaRepo.filter("rin"))
				.expectNext("System.string")
				.verifyComplete();

		StepVerifier.create(schemaRepo.filter("ny"))
				.expectNext("System.any")
				.verifyComplete();

		StepVerifier.create(schemaRepo.filter("").sort().collect(Collectors.toSet()))
				.expectNext(Set.of("System.Date.Duration", "System.Date.Timeunit", "System.Null", "System.float",
						"System.number", "System.Date.TimeObject", "System.ParameterExpression",
						"System.Date.Timestamp", "System.long", "System.integer", "System.string",
						"System.Schema", "System.boolean", "System.double", "System.any"))
				.verifyComplete();
	}

}